src/wallet/mod.rs

```diff
@@ -753,13 +751,25 @@ where
                 script_pubkey: script_pubkey.clone(),
                 value,
             };
-            fee_amount += fee_rate.fee_vb(serialize(&new_out).len());
 
             tx.output.push(new_out);
 
             outgoing += value;
         }
 
+        fee_amount += fee_rate.fee_wu(tx.weight());
```

Comment:
An interesting thing to note here: while rust-bitcoin serializes txs with no inputs as segwit txs (hence, including the marker and flag bytes), the `Transaction::weight()` method doesn't!

I opened an issue to point out the inconsistency: https://github.com/rust-bitcoin/rust-bitcoin/issues/1159

This is just to say that the current code that explicitly pushes the two extra weight units is actually correct, but this may change in the future if the `weight` method ends up being modified.